--- src/plugins/codecs/kr/cp949codetbl.h.orig	2014-04-10 12:37:12.000000000 -0600
+++ src/plugins/codecs/kr/cp949codetbl.h	2014-08-19 19:40:07.000000000 -0600
@@ -40,7 +40,7 @@
 ****************************************************************************/
 
 #ifndef CP949CODETBL_H
-#define CP494CODETBL_H
+#define CP949CODETBL_H
 
 static const unsigned short cp949_icode_to_unicode[] = {
 0xac02, 0xac03, 0xac05, 0xac06, 0xac0b, 0xac0c, 0xac0d, 0xac0e, 0xac0f, 0xac18, 0xac1e, 0xac1f, 0xac21, 0xac22, 0xac23, 
